From f726197117b7b2d34ab7b3120e9825b45d79d4a8 Mon Sep 17 00:00:00 2001
From: Huibin Hong <huibin.hong@rock-chips.com>
Date: Tue, 19 Dec 2017 15:13:40 +0800
Subject: [PATCH] fiq_debugger: rockchip: move uart_init to the last

1.When fiq debugger driver probes, and uart rx is pulled low.
The uart interrupt comes so early that uart handler can't be
called. Which makes interrupt come frequently and block the cpu.
2.Remove '<hit enter to activate fiq debugger>'.

Change-Id: I4656d6942bb2cbb0e992852a52bc82f71deadda8
Signed-off-by: Huibin Hong <huibin.hong@rock-chips.com>
---
 .../android/fiq_debugger/fiq_debugger.c       | 22 ++++++++-----------
 1 file changed, 9 insertions(+), 13 deletions(-)

diff --git a/drivers/staging/android/fiq_debugger/fiq_debugger.c b/drivers/staging/android/fiq_debugger/fiq_debugger.c
index e9a2a3e19a6a..be7ef2512610 100644
--- a/drivers/staging/android/fiq_debugger/fiq_debugger.c
+++ b/drivers/staging/android/fiq_debugger/fiq_debugger.c
@@ -1401,16 +1401,6 @@ static int fiq_debugger_probe(struct platform_device *pdev)
 	if (state->clk)
 		clk_enable(state->clk);
 
-	if (pdata->uart_init) {
-		ret = pdata->uart_init(pdev);
-		if (ret)
-			goto err_uart_init;
-	}
-
-	fiq_debugger_printf_nfiq(state,
-				"<hit enter %sto activate fiq debugger>\n",
-				state->no_sleep ? "" : "twice ");
-
 	if (fiq_debugger_have_fiq(state)) {
 #ifdef CONFIG_FIQ_GLUE
 #ifdef CONFIG_FIQ_DEBUGGER_TRUST_ZONE
@@ -1451,9 +1441,6 @@ static int fiq_debugger_probe(struct platform_device *pdev)
 		enable_irq_wake(state->uart_irq);
 	}
 
-	if (state->clk)
-		clk_disable(state->clk);
-
 	if (state->signal_irq >= 0) {
 		ret = request_irq(state->signal_irq, fiq_debugger_signal_irq,
 			  IRQF_TRIGGER_RISING, "debug-signal", state);
@@ -1486,6 +1473,15 @@ static int fiq_debugger_probe(struct platform_device *pdev)
 	state_tf = state;
 #endif
 
+	if (pdata->uart_init) {
+		ret = pdata->uart_init(pdev);
+		if (ret)
+			goto err_uart_init;
+	}
+
+	if (state->clk)
+		clk_disable(state->clk);
+
 #if defined(CONFIG_FIQ_DEBUGGER_CONSOLE)
 	spin_lock_init(&state->console_lock);
 	state->console = fiq_debugger_console;
-- 
2.35.3

